### PR TITLE
Fixes for mesh expansion

### PR DESCRIPTION
**Pull request by costinm**
In 1.0, the mesh expansion is using the Gateway IP, with 2 mtls ports (CA and pilot).

Instead of overriding /etc/hosts or using dnsmasq we can set the IP in the istio config ( cluster.env) - 
for that to work we need node_agent to set the proper expected ServerName for validation. 
Since we may run pilot replicas on-prem we also need node_agent to be able to renew pilot cert
which also has a DNS name. 

Added the options to CA so pilot is provisioned with a DNS SAN ( there was also a bug in the code )

### CODE REVIEW

File: security/pkg/platform/onprem.go [hunk 1]
```diff
@@ -84,6 +89,11 @@ func (ci *OnPremClientImpl) GetServiceIdentity() (string, error) {
 		return "", err
 	}
 	if len(serviceIDs) != 1 {
+		for _, s := range serviceIDs {
+			if strings.HasPrefix(s, "spiffe://") {
+				return serviceIDs[0], nil
+			}
+		}
 		return "", fmt.Errorf("cert has %v SAN fields, should be 1", len(serviceIDs))
```
Comment (myidpt) [inline security/pkg/platform/onprem.go]: Change the error message? I.e., "Cert does not have SPIFFE identity in SAN"

File: security/pkg/platform/onprem.go [hunk 2]
```diff
@@ -84,6 +89,11 @@ func (ci *OnPremClientImpl) GetServiceIdentity() (string, error) {
 		return "", err
 	}
 	if len(serviceIDs) != 1 {
+		for _, s := range serviceIDs {
+			if strings.HasPrefix(s, "spiffe://") {
+				return serviceIDs[0], nil
```
Comment (myidpt) [inline security/pkg/platform/onprem.go]: Return s instead of serviceIDs[0]?

Comment (myidpt) [reply security/pkg/platform/onprem.go]: This function gets the identity from the service (on VM) cert, which should be single value. Do we want to change this?

File: security/pkg/platform/onprem.go (hunk 1 above)
Comment (costinm) [reply security/pkg/platform/onprem.go]: done

File: security/pkg/platform/onprem.go (hunk 2 above)
Comment (costinm) [reply security/pkg/platform/onprem.go]: It is not single value for pilot and other components that need to interoperate with non-spiffe clients.

Comment (myidpt) [reply security/pkg/platform/onprem.go]: Oh I see. You are right.

**Pull request merged**